### DISCUSSION TITLE
How to type-check `graphql` operations in a CI/build step

### DISCUSSION THREAD

**Discussion by WonderPanda** [0no-co/gql.tada#344] [Q&A] [answered]
Apologies if this is answered somewhere but I haven't been able to figure it out yet.

If I have a GraphQL operation that is not valid, it is correctly identified in my IDE (VsCode). However, I haven't been able to figure out a way to include a build/typecheck step in CI or locally which will flag these issues via `tsc`.

We recently hit an issue where our CI passed after a refactor to our GraphQL schema which resulted in invalid operations defined in app code.

Eg:
![image](https://github.com/user-attachments/assets/ad607b29-a10d-4670-af8d-6b09f3205499)

There operation is invalid because there is an additional required field in `pkColumns`. My IDE clearly knows there is a problem but I would like to have this fail the build

**Comment by JoviDeCroock** [comment, marked as answer]
Our CLI had a check command https://gql-tada.0no.co/reference/gql-tada-cli and the CI workflow is documented in our docs as well

**Reply by WonderPanda** [reply]
Ahhh thank you. Sorry I missed it, this is great
